### PR TITLE
ROCm enable sparse_sampled_addmm

### PR DESCRIPTION
**Pull request by jpvillam-amd**
Enables:
test_comprehensive_sparse_sampled_addmm_cuda_complex128
test_comprehensive_sparse_sampled_addmm_cuda_complex64
test_comprehensive_sparse_sampled_addmm_cuda_float32
test_comprehensive_sparse_sampled_addmm_cuda_float64
test_dispatch_meta_sparse_sampled_addmm_cuda_complex128
test_dispatch_meta_sparse_sampled_addmm_cuda_complex64
test_dispatch_meta_sparse_sampled_addmm_cuda_float32
test_dispatch_meta_sparse_sampled_addmm_cuda_float64
test_meta_sparse_sampled_addmm_cuda_complex128
test_meta_sparse_sampled_addmm_cuda_complex64
test_meta_sparse_sampled_addmm_cuda_float32
test_meta_sparse_sampled_addmm_cuda_float64

### CODE REVIEW

**Review by pruthvistony — APPROVED** [review]
The changes are fine.
Waiting for CI run to check if above tests are executed successfully.

**Comment by pruthvistony** [comment]
Oct 11 19:57:04   test_comprehensive_sparse_sampled_addmm_cuda_complex128 (__main__.TestDecompCUDA) ... ok (0.164s)
Oct 11 19:57:04   test_comprehensive_sparse_sampled_addmm_cuda_complex64 (__main__.TestDecompCUDA) ... ok (0.071s)
Oct 11 19:57:04   test_comprehensive_sparse_sampled_addmm_cuda_float32 (__main__.TestDecompCUDA) ... ok (0.054s)
Oct 11 19:57:04   test_comprehensive_sparse_sampled_addmm_cuda_float64 (__main__.TestDecompCUDA) ... ok (0.056s)

Oct 11 20:33:09   test_dispatch_meta_sparse_sampled_addmm_cuda_complex128 (__main__.TestMetaCUDA) ... ok (0.026s)
Oct 11 20:33:09   test_dispatch_meta_sparse_sampled_addmm_cuda_complex64 (__main__.TestMetaCUDA) ... ok (0.008s)
Oct 11 20:33:09   test_dispatch_meta_sparse_sampled_addmm_cuda_float32 (__main__.TestMetaCUDA) ... ok (0.008s)
Oct 11 20:33:09   test_dispatch_meta_sparse_sampled_addmm_cuda_float64 (__main__.TestMetaCUDA) ... ok (0.007s)

Oct 11 20:37:11   test_meta_sparse_sampled_addmm_cuda_complex128 (__main__.TestMetaCUDA) ... ok (0.008s)
Oct 11 20:37:11   test_meta_sparse_sampled_addmm_cuda_complex64 (__main__.TestMetaCUDA) ... ok (0.008s)
Oct 11 20:37:11   test_meta_sparse_sampled_addmm_cuda_float32 (__main__.TestMetaCUDA) ... ok (0.008s)
Oct 11 20:37:11   test_meta_sparse_sampled_addmm_cuda_float64 (__main__.TestMetaCUDA) ... ok (0.007s)

File: torch/testing/_internal/common_cuda.py [hunk 1]
```diff
@@ -173,6 +173,13 @@ def _get_torch_cuda_version():
     cuda_version = str(torch.version.cuda)
     return tuple(int(x) for x in cuda_version.split("."))
 
+def _get_torch_hip_version():
```
Comment (pruthvistony) [inline torch/testing/_internal/common_cuda.py:176]: NIT comment - better to name this function `_get_torch_rocm_version`

**Review by pruthvistony — CHANGES_REQUESTED** [review]

**Comment by jpvillam-amd** [comment]
Not really sure why the CI failed here. Could anyone help me figure it out?

**Comment by pruthvistony** [comment]
jenkins retest this please

**Comment by pruthvistony** [comment]
jenkins retest this please

**Comment by pruthvistony** [comment]
> Not really sure why the CI failed here. Could anyone help me figure it out?

I also didnt get why it was failing, then realized it is due to git merge conflict. Resolved it.

**Comment by pruthvistony** [comment]
The tests are passing
Oct 14 04:35:18   test_comprehensive_sparse_sampled_addmm_cuda_complex128 (__main__.TestDecompCUDA) ... ok (0.146s)
Oct 14 04:35:18   test_comprehensive_sparse_sampled_addmm_cuda_complex64 (__main__.TestDecompCUDA) ... ok (0.063s)
Oct 14 04:35:18   test_comprehensive_sparse_sampled_addmm_cuda_float32 (__main__.TestDecompCUDA) ... ok (0.052s)
Oct 14 04:35:18   test_comprehensive_sparse_sampled_addmm_cuda_float64 (__main__.TestDecompCUDA) ... ok (0.051s)

**Review by pruthvistony — APPROVED** [review]

**Pull request merged**